src/config.rs

```diff
@@ -133,9 +131,10 @@ impl Cfg {
         Ok(())
     }
 
-    pub fn set_profile(&self, profile: &str) -> Result<()> {
-        if !dist::Profile::names().contains(&profile) {
-            return Err(ErrorKind::UnknownProfile(profile.to_owned()).into());
+    pub fn set_profile(&mut self, profile: &str) -> Result<()> {
+        let p = dist::Profile::from_str(profile)?;
+        if self.profile_override.is_some() {
+            self.profile_override = Some(p)
         }
```

Comment:
I wonder, is it right to unset the profile override here?  To be fair, `set_profile()` is only called in a small number of pathways and probably it doesn't matter; but it does feel odd to see this change.

I think if anything, setting the profile explicitly should probably just unset the override (set it to `None`) -- what do you think?